pkg/reconciler/workload/placement/placement_reconcile_scheduling.go

```diff
@@ -116,12 +118,66 @@ func (r *placementSchedulingReconciler) getAllValidSyncTargetsForPlacement(clust
 		return locationWorkspace, nil, err
 	}
 
+	locationClusters, err = r.filterAPICompatible(ctx, clusterName, locationClusters)
+	if err != nil {
+		return locationWorkspace, nil, err
+	}
+
 	// find all the valid sync targets.
 	validClusters := locationreconciler.FilterNonEvicting(locationreconciler.FilterReady(locationClusters))
 
 	return locationWorkspace, validClusters, nil
 }
 
+func (r *placementSchedulingReconciler) filterAPICompatible(ctx context.Context, clusterName logicalcluster.Name, syncTargets []*workloadv1alpha1.SyncTarget) ([]*workloadv1alpha1.SyncTarget, error) {
```

Comment:
Do you think we could have a way to surface to the Placement conditions the fact that no SyncTarget for the given location provides the expected APIs (with the list of missing APIs) ?
That would make debugging for the end-user much easier ?